tests/test_observables.py

```diff
@@ -19,29 +19,111 @@
+
+    _rbm_am = None
+    _device = None
+
+    def __init__(self, nqubits, state=None):
+        self.nqubits = nqubits
+        self.device = "cpu"
+
+    def sample(self, num_samples=1):
+        dist = torch.distributions.Bernoulli(probs=0.5)
+        sample_size = torch.Size((num_samples, self.nqubits))
+        initial_state = dist.sample(sample_size).to(
+            device=self.device, dtype=torch.double
+        )
+
+        return initial_state
+
+    @staticmethod
+    def autoload(location, gpu=False):
+        pass
+
+    def gradient(self, v):
+        return 0.0
+
+    @property
+    def rbm_am(self):
+        return self._rbm_am
+
+    @rbm_am.setter
+    def rbm_am(self, new_val):
+        self._rbm_am = new_val
+
+    @property
+    def device(self):
+        return self._device
+
+    @device.setter
+    def device(self, new_val):
+        self._device = new_val
+
+    @property
+    def networks(self):
+        return ["rbm_am"]
+
+    def phase(self, v):
+        if v.dim() == 1:
+            v = v.unsqueeze(0)
+            unsqueezed = True
+        else:
+            unsqueezed = False
+
+        phase = torch.zeros(v.shape[0])
+
+        if unsqueezed:
+            return phase.squeeze_(0)
+        else:
+            return phase
+
+    def amplitude(self, v):
+        return 1 / torch.sqrt(torch.tensor(float(self.nqubits))) * torch.ones(v.size(0))
+
+    def psi(self, v):
+        # vector/tensor of shape (len(v),)
+        amplitude = self.amplitude(v)
+
+        # complex vector; shape: (2, len(v))
+        psi = torch.zeros((2,) + amplitude.shape).to(
+            dtype=torch.double, device=self.device
+        )
+        psi[0] = amplitude
+
+        # squeeze down to complex scalar if there was only one visible state
+        return psi.squeeze()
+
 
 # TODO: add assertions
 
 
 class TestPauli(unittest.TestCase):
     def test_spinflip(self):
-        test_psi = ComplexWaveFunction(2, num_hidden=3)
-        test_sample = test_psi.sample(100, num_samples=1000)
+        test_psi = TestWaveFunction(2)
+        test_sample = test_psi.sample(num_samples=1000)
         observables.pauli.flip_spin(1, test_sample)
 
     def test_apply(self):
-        test_psi = ComplexWaveFunction(2, num_hidden=3)
-        test_sample = test_psi.sample(100, num_samples=1000)
+        test_psi = TestWaveFunction(2)
+        test_sample = test_psi.sample(num_samples=1000)
         X = observables.SigmaX()
-        X.apply(test_psi, test_sample)
+
+        measure_X = float(X.apply(test_psi, test_sample).mean())
+        self.assertTrue(1.0 == measure_X, msg="measure Pauli X failed")
 
         Y = observables.SigmaY()
-        Y.apply(test_psi, test_sample)
+        measure_Y = float(Y.apply(test_psi, test_sample).mean())
+        self.assertTrue(0.0 == measure_Y, msg="measure Pauli Y failed")
 
         Z = observables.SigmaZ()
-        Z.apply(test_psi, test_sample)
+        measure_Z = float(Z.apply(test_psi, test_sample).mean())
```

Comment:
here, your samples are in the 0/1 format, but the magnetization observables (i think all of them, really) expect samples to be in +/-1 format

you can use the `to_pm1` function in `observables.utils` to convert to +/- 1 format right after you draw the samples

let me know if this breaks the other tests (i don't think it should though)

Reply:
But in `PauliZ`:

```
    def apply(self, nn_state, samples):
        r"""Computes the magnetization of each sample given a batch of samples.

        :param nn_state: The WaveFunction that drew the samples.
        :type nn_state: qucumber.nn_states.WaveFunction
        :param samples: A batch of samples to calculate the observable on.
                        Must be using the :math:`\sigma_i = 0, 1` convention.
        :type samples: torch.Tensor
        """
        # convert to +/- 1 convention, after computing the
        # mean, to reduce total computations; this works
        # because expectation is linear.
        return to_pm1(samples.mean(1)).abs()
```

The comments mean the input is 0/1 format?